test/mobile/nnc/test_context.cpp

```diff
@@ -0,0 +1,156 @@
+#include <gtest/gtest.h>
+#include <torch/csrc/jit/mobile/nnc/context.h>
+#include <torch/csrc/jit/mobile/nnc/registry.h>
+#include <ATen/Functions.h>
+
+namespace torch {
+namespace jit {
+namespace mobile {
+namespace nnc {
+
+extern "C" {
+
+// out = a * n (doing calculation in the `tmp` buffer)
+int bad_mul_kernel(void** args) {
+  const int size = 128;
+  at::Tensor a = at::from_blob(args[0], {size}, at::kFloat);
+  at::Tensor out = at::from_blob(args[1], {size}, at::kFloat);
+  at::Tensor n = at::from_blob(args[2], {1}, at::kInt);
+  at::Tensor tmp = at::from_blob(args[3], {size}, at::kFloat);
+
+  tmp.zero_();
+  for (int i = n.item().toInt(); i > 0; i--) {
+    tmp.add_(a);
+  }
+  out.copy_(tmp);
+  return 0;
+}
+
+int dummy_kernel(void** /* args */) {
+  return 0;
+}
+
+} // extern "C"
+
+REGISTER_NNC_KERNEL("bad_mul", bad_mul_kernel)
+REGISTER_NNC_KERNEL("dummy", dummy_kernel)
+
+InputSpec create_test_input_spec(const std::vector<int64_t>& sizes) {
+  InputSpec input_spec;
+  input_spec.sizes_ = sizes;
+  input_spec.dtype_ = at::kFloat;
+  return input_spec;
+}
+
+OutputSpec create_test_output_spec(const std::vector<int64_t>& sizes) {
+  OutputSpec output_spec;
+  output_spec.sizes_ = sizes;
+  output_spec.dtype_ = at::kFloat;
+  return output_spec;
+}
+
+MemoryPlan create_test_memory_plan(const std::vector<int64_t>& buffer_sizes) {
+  MemoryPlan memory_plan;
+  memory_plan.buffer_sizes_ = buffer_sizes;
```

Comment:
Should memory plan have API where it can create buffer sizes off of list of tensors?